src/App.js

```diff
@@ -1,33 +1,49 @@
-import React from 'react';
-import restaurants from 'data/restaurants.json';
+import React, { useState, useEffect } from 'react';
 import CardUi from 'components/CardUi';
 import NavBar from 'components/Navbar';
 import 'styles/App.scss';
+import { loadRestaurantData } from 'App.utils';
 
-function deepCopy(data) {
-  return JSON.parse(JSON.stringify(data));
-}
 
-function ShuffleArr(input) {
-  for (let i = input.length - 1; i >= 0; i -= 1) {
-    const randomIndex = Math.floor(Math.random() * (i + 1));
-    const itemAtIndex = input[randomIndex];
+const App = () => {
+  const [dataRequest, setDataRequest] = useState({
+    data: [],
+    isLoading: true,
+    error: null,
+  });
 
-    input[randomIndex] = input[i];
-    input[i] = itemAtIndex;
+  useEffect(() => {
+    loadRestaurantData({ setDataRequest });
+  }, []);
+
+  function deepCopy(data) {
+    return JSON.parse(JSON.stringify(data));
+  }
+
+  function ShuffleArr(input) {
+    for (let i = input.length - 1; i >= 0; i -= 1) {
+      const randomIndex = Math.floor(Math.random() * (i + 1));
+      const itemAtIndex = input[randomIndex];
+
+      input[randomIndex] = input[i];
+      input[i] = itemAtIndex;
+    }
+    return input;
   }
-  return input;
-}
 
-const cardCycle = ShuffleArr(deepCopy(restaurants));
+  const cardCycle = ShuffleArr(deepCopy(dataRequest.data));
 
-function App() {
   return (
     <>
-      <NavBar />
-      <CardUi list={cardCycle} />
+      {dataRequest.isLoading && <div>*Insert loading spinner here...*</div>}
```

Comment:
I want to add a generic spinner here before merge happens so we can get that done, let me know if you have any objections or if you'd like me to wait until this is merged.